contracts/NFTVesting/LinearVestingNFTTimeLock.sol

```diff
@@ -0,0 +1,181 @@
+        );
+        require(
+            address(this).balance > 0,
+            "Time:Lock: Eth should be sent to contract before initialization"
+        );
+
+        _nft = nft_;
+        _tokenId = tokenId_;
+        _beneficiary = beneficiary_;
+        _vestingStartTime = vestingStartTime_;
+        _maxDiscountPercentage = maxDiscountPercentage_;
+        _maxDuration = maxDuration_;
+    }
+
+    /**
+     * @dev Returns the smart contract NFT.
+     */
+    function nft() public view virtual returns (IERC721) {
+        return _nft;
+    }
+
+    /**
+     * @dev Returns the token ID of the NFT being held.
+     */
+    function tokenId() public view virtual returns (uint256) {
+        return _tokenId;
+    }
+
+    /**
+     * @dev Returns the beneficiary that will receive the NFT.
+     */
+    function beneficiary() public view virtual returns (address) {
+        return _beneficiary;
+    }
+
+    /**
+     * @dev Returns the time when the NFT are released in seconds since Unix epoch (i.e. Unix timestamp).
+     */
+    function vestingStartTime() public view virtual returns (uint256) {
+        return _vestingStartTime;
+    }
+
+    /**
+     * @dev Returns the max duration that a token is allowed to vest
+     */
+    function maxDuration() public view virtual returns (uint256) {
+        return _maxDuration;
+    }
+
+    /**
+     * @dev Returns the max discount allowed for a token in percentage.
+     */
+    function maxDiscountPercentage() public view virtual returns (uint256) {
+        return _maxDiscountPercentage;
+    }
+
+    /**
+     * @dev Returns duration that NFT has been locked and vesting
+     */
+    function vestedDuration() public view returns (uint256) {
+        return block.timestamp - vestingStartTime();
+    }
+
+    /**
+     * @dev Returns discount percentage for achieved from vesting
+     * Based off: discount = mx
+     */
+    function getDiscountPercentage() public view returns (uint256) {
+        if (block.timestamp < vestingStartTime()) {
+            return 0;
+        }
+        return (maxDiscountPercentage() * vestedDuration()) / maxDuration();
+    }
+
+    /**
+     * @dev Returns discount accrued in Eth according to duration vested
+     */
+    function getDiscount() public view returns (uint256) {
+        return address(this).balance * getDiscountPercentage();
+    }
+
+    /**
+     * @dev Transfers NFT held by the timelock to the beneficiary. Will only succeed if invoked after the release
+     * time. Sends the discount in Eth to the beneficiary.
+     * Reverts if transfer of NFT fails.
+     */
+    function release() public virtual {
+        require(
+            block.timestamp >= vestingStartTime(),
+            "TimeLock: current time is before vesting start time"
+        );
+        require(
+            nft().ownerOf(tokenId()) == address(this),
+            "TimeLock: no NFT to release for this address"
+        );
+
+        uint256 ethDiscount = getDiscount();
```

Comment:
what happens if getDiscount() returns a zero? i see that there is a check for `block.timestamp >= vestingStartTime()` but since getDiscountPercentage() returns a rounded down whole number, there is a chance that it returns zero

Reply:
thanks @DominiqueYeo for the spot! It could return 0, but that is only if block.timestamp < vesting start time. once it starts vesting, in `getDiscountPercentage`         it will `return (maxDiscountPercentage() * vestedDuration()) / maxDuration();`

Therefore it will only return 0 if vestedDuration() is 0 or max DiscountPercentage is 0


So i think its not necessary to add a check, because if it returns 0 its by design from the developers?